neuro_data/static_images/dataset_config.py

```diff
@@ -309,6 +398,105 @@ def stats(self, condition_hashes, images, responses, tiers):
             statistics = dict(images=input_statistics, responses=response_statistics)
             return statistics
 
+    class NoBehFullFieldFrame2(dj.Part):
+        # implemented only for full field stimulus with stimulus_type=='stimulus.Frame2'
+        definition = """
+        # mimic the behavior of InputResponse with the corresponding preprocess_id to compute the statistics of input and responses, do not compute stats for behavior variables
+        -> master
+        ---
+        stats_tier="train"                 : enum("train", "test", "validation", "all")               # tier used for computing stats
+        stats_per_input                    : tinyint                                                  # whether to compute stats per input
+        """
+        content = [
+            dict(stats_tier="train", stats_per_input=0),
+        ]
+
+        @staticmethod
+        def run_stats_resp(data, ix, axis=0):
+            ret = {}
+            data = data[ix]
+            ret["all"] = dict(
+                mean=data.mean(axis=axis).astype(np.float32),
+                std=data.std(axis=axis, ddof=1).astype(np.float32),
+                min=data.min(axis=axis).astype(np.float32),
+                max=data.max(axis=axis).astype(np.float32),
+                median=np.median(data, axis=axis).astype(np.float32),
+            )
+            ret["stimulus.Frame2"] = ret["all"]
+            return ret
+
+        def run_stats_input(self, data, info_df, ix, per_input):
+            if per_input:
+                raise NotImplementedError("per_input is not implemented for Frame2")
+            ret = {}
+            data = data[ix]
+            info_df = info_df.loc[ix]
+            from statsmodels.stats.weightstats import DescrStatsW as wstats
+
+            sizes = [d.squeeze().shape for d in data]
+            assert len(set(sizes)) == 1, "All images must have the same size"
+            masks = [
+                frame2_make_mask(
+                    row["aperture_x"],
+                    row["aperture_y"],
+                    row["aperture_r"],
+                    row["aperture_transition"],
+                    sizes[0],
+                )
+                for _, row in info_df.iterrows()
+            ]
+            data_flat = np.hstack([d.flatten() for d in data])
+            data_mask = np.hstack([m.flatten() for m in masks])
+
+            data_mean = wstats(data_flat, data_mask).mean
+            data_std = wstats(data_flat, data_mask).std
+
+            ret["stimulus.Frame2"] = dict(
+                mean=data_mean.astype(np.float32),
+                std=data_std.astype(np.float32),
+                min=data.min().astype(np.float32),
+                max=data.max().astype(np.float32),
+                median=np.median(data).astype(np.float32),
```

Comment:
median isn't computed with correct masking. downstream computation shouldn't use this value.

Reply:
Comment added.